--- configure.ac.ori	2012-09-21 11:02:24.810308571 +0200
+++ configure.ac	2012-09-21 11:02:50.922307637 +0200
@@ -1553,7 +1553,7 @@
     CFLAGS="$CFLAGS $pplinc $gmpinc"
     AC_MSG_CHECKING([for version 0.11 (revision 0 or later) of PPL])
     AC_TRY_COMPILE([#include "ppl_c.h"],[
-    #if PPL_VERSION_MAJOR != 0 || PPL_VERSION_MINOR < 11
+    #if PPL_VERSION_MAJOR == 0 && PPL_VERSION_MINOR < 11
     choke me
     #endif
     ], [AC_MSG_RESULT([yes])], [AC_MSG_RESULT([no]); ppllibs= ; pplinc= ; with_ppl=no ])
