--- build-gcc-arm.sh	2010-12-20 13:01:15.000000000 +0100
+++ my_build-gcc-arm.sh	2010-12-20 12:58:49.000000000 +0100
@@ -33,10 +33,10 @@
 set -o errexit
 set -o pipefail
 
-#export PATH=${HOME}/bin:${PATH}
-#export LD_LIBRARY_PATH=${HOME}/lib:${LD_LIBRARY_PATH}
-#echo $PATH
-#echo $LD_LIBRARY_PATH
+export PATH=${HOME}/bin:${PATH}
+export LD_LIBRARY_PATH=${HOME}/lib:${LD_LIBRARY_PATH}
+echo $PATH
+echo $LD_LIBRARY_PATH
 
 CORTEX_TOPDIR=`pwd`
 
@@ -77,7 +77,7 @@
 TOOLCHAIN_TARGET=arm-lancos-eabi
 
 #Numero di compilazioni concorrenti (consigliabile 2+ per un dual-core o 4+ per un quad-core)
-NUM_JOBS=2
+NUM_JOBS=4
 
 mkdir -p ${TOOLCHAIN_PATH}
 
@@ -282,7 +282,7 @@
 	cd binutils-${BINUTILS_VER}
 
 	# hack: allow autoconf version 2.65 instead of 2.64
-	sed -i 's@\(.*_GCC_AUTOCONF_VERSION.*\)2.64\(.*\)@\12.65\2@' config/override.m4
+	#sed -i 's@\(.*_GCC_AUTOCONF_VERSION.*\)2.64\(.*\)@\12.65\2@' config/override.m4
 	autoconf
 	mkdir build
 	cd build
@@ -323,7 +323,7 @@
 	#cd ..
 
 	# hack: allow autoconf version 2.65 instead of 2.64
-	sed -i 's@\(.*_GCC_AUTOCONF_VERSION.*\)2.64\(.*\)@\12.65\2@' config/override.m4
+	#sed -i 's@\(.*_GCC_AUTOCONF_VERSION.*\)2.64\(.*\)@\12.65\2@' config/override.m4
 	autoconf
 
 	mkdir build
@@ -366,7 +366,7 @@
 	cd newlib-${NEWLIB_VER}
 
 	# hack: allow autoconf version 2.65 instead of 2.64
-	sed -i 's@\(.*_GCC_AUTOCONF_VERSION.*\)2.64\(.*\)@\12.65\2@' config/override.m4
+	#sed -i 's@\(.*_GCC_AUTOCONF_VERSION.*\)2.64\(.*\)@\12.65\2@' config/override.m4
 	autoconf
 	mkdir build
 	cd build
