--- newlib/libc/include/sys/time.h	2018-02-26 20:39:48.000000000 +0100
+++ newlib/libc/include/sys/new_time.h	2018-03-30 14:48:34.967448212 +0200
@@ -274,8 +274,8 @@
 int adjtime (const struct timeval *, struct timeval *);
 int futimes (int, const struct timeval *);
 int lutimes (const char *, const struct timeval *);
-int settimeofday (const struct timeval *, const struct timezone *);
 #endif
+int settimeofday (const struct timeval *, const struct timezone *);
 
 #if __MISC_VISIBLE || __XSI_VISIBLE
 int getitimer (int __which, struct itimerval *__value);
