--- newlib-2.1.0/newlib/libc/locale/lctype.c	2010-05-03 11:25:32.000000000 +0200
+++ newlib-2.1.0_my/newlib/libc/locale/lctype.c	2014-11-05 01:16:36.831143438 +0100
@@ -95,9 +95,9 @@
 		_ctype_locale_buf, "LC_CTYPE",
 		LCCTYPE_SIZE, LCCTYPE_SIZE,
 		(const char **)&_ctype_locale);
-	if (ret == 0 && _ctype_using_locale)
-		_ctype_locale.grouping =
-			__fix_locale_grouping_str(_ctype_locale.grouping);
+//	if (ret == 0 && _ctype_using_locale)
+//		_ctype_locale.grouping =
+//			__fix_locale_grouping_str(_ctype_locale.grouping);
 #endif
 	return ret;
 }
