--- newlib-2.1.0/newlib/libc/locale/locale.c	2012-03-29 22:27:15.000000000 +0200
+++ newlib-2.1.0_my/newlib/libc/locale/locale.c	2014-11-05 00:40:29.952398466 +0100
@@ -466,6 +466,7 @@
   int (*l_mbtowc) (struct _reent *, wchar_t *, const char *, size_t,
 		   const char *, mbstate_t *);
   int cjknarrow = 0;
+  int ret = 0;
 
   /* Avoid doing everything twice if nothing has changed. */
   if (!strcmp (new_categories[category], current_categories[category]))
